Integrate dev into main

weissercn opened the pull request.

weissercn commented:
Currently failing due to the following error. Investigating
File "/nas/ucb/constantinweisser/Influence-benchmark/influence_benchmark/RL/expert_iteration.py", line 148, in launch
    selected_trajectories = get_top_n_trajectories(trajectory_folder, self.num_chosen_trajectories, self.mode)

carolius commented on influence_benchmark/experiments/hyperparameter_sweep.py:
```diff
@@ -52,7 +52,7 @@ def train_loop(config=None):
             sft_script_path=sft_script_path,
             model_name=model_name,
-            num_gen_trajectories_per_state=config.num_gen_trajectories_per_state,
+            num_gen_trajectories_per_state=6,
```
kind of depends which parameters we want to vary each run but we might not want to push version with 6 hardcoded

micahcarroll replied:
I guess we're doing this?

carolius commented:
We might want to check if sft trainer still correctly trains on the old lora rather than generating a new one after your changes, it can behave strangely and fail silently in my experience

weissercn commented:
Done
- added environment queue
- added comments 
- fixed sbatch issues (needed to specify 1 node)
- removing code redundancies
- added slurm preparation
- saving settings for run 
- changed some copy to deepcopy

micahcarroll commented on influence_benchmark/experiments/KTO_experiment copy.py:
Do we want to merge a `KTO_experiment copy.py` file? Probably not

micahcarroll commented on influence_benchmark/experiments/expert_iteration_experiment.py:
```diff
@@ -21,15 +21,24 @@
     ignore_first_n_assistant_messages = 1  # Number of assistant messages to not train on
     run_name = None
-    devices = [0, 1, 2, 3]  # , 4, 5, 6, 7]
+    devices = [
+        0,
+        1,
+        2,
+        3,
+    ]  # GPUs used for generating trajectories. The GPUs used for training are specified in the accelerate_config.yaml file.
```
```suggestion
    # GPUs used for generating trajectories. The GPUs used for training are specified in the accelerate_config.yaml file.
    devices = [0, 1, 2, 3]  
```

micahcarroll commented on influence_benchmark/experiments/single_expert_iteration_experiment.py:
```diff
@@ -8,16 +8,19 @@
     run_name = None
-    devices = [1]
-    mode = "single"  # single environment
+    devices = [
+        2
+    ]  # GPUs used for generating trajectories. The GPUs used for training are specified in the accelerate_config.yaml file.
```
```suggestion
    # GPUs used for generating trajectories. The GPUs used for training are specified in the accelerate_config.yaml file.
    devices = [2]  
```

micahcarroll commented on influence_benchmark/stats/preferences_per_iteration.py:
```diff
@@ -1,86 +1,95 @@
+    if "initial_state_id" not in trajectories.columns:
+        trajectories["initial_state_id"] = 0
+
+    # Average over turns
+    # Group by env_name, initial_state_id, and trajectory_id, and calculate average reward
+    avg_rewards = (
+        trajectories.groupby(["env_name", "initial_state_id", "trajectory_id"])["expected_preference"]
+        .mean()
+        .reset_index()
+    )
+
+    # Select top N trajectories for each env_name and initial_state_id
+    top_n = (
+        avg_rewards.groupby(["env_name", "initial_state_id"])
+        .apply(
+            lambda x: x.assign(
+                n_trajectories=len(x), reward_avg_all_trajectories=x["expected_preference"].mean()
+            ).nlargest(num_chosen_trajectories, "expected_preference")
+        )
+        .reset_index(drop=True)
+    )
+    top_n = top_n.rename(
+        columns={
+            "expected_preference": "reward_avg_selected_trajectories",  # average after selecting trajectories # nlargest_[trajectory_id](mean_[turn](reward))
+        }
+    )
```
Not carefully reviewed, just going to assume it's correct

micahcarroll reviewed (approved):
LGTM! Let's wait to merge until the remaining SLURM stuff bugs are figured out and merged
